crates/sim2h/src/crypto.rs

```diff
@@ -0,0 +1,211 @@
+//! copied from holochain_rust::dpki::utils for sim2h server
+
+use crate::{
+    error::{Sim2hError, Sim2hResult},
+    wire_message::WireMessage,
+};
+use hcid::*;
+use holochain_core_types::{
+    agent::Base32,
+    error::{HcResult, HolochainError},
+    signature::Provenance,
+};
+use holochain_persistence_api::cas::content::Address;
+use lib3h_protocol::data_types::Opaque;
+use lib3h_sodium::{secbuf::SecBuf, sign};
+
+use std::convert::TryFrom;
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct SignedWireMessage {
+    pub provenance: Provenance,
+    pub payload: Opaque,
+}
+
+impl SignedWireMessage {
+    pub fn new(
+        mut secret_key: &mut SecBuf,
```

Comment:
Not sure if we can use this constructor signature in the worker. The key is held in the keystore, behind the conductor api. I guess we will have to send the payload to the conductor api and receive back the signature. Which we will need a constructor for SignedWireMessage that just receives payload and signature..